pkg/webhooks/kubedb/v1alpha2/hazelcast.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("podTemplate").Child("spec").Child("containers"),
+			h.Name,
+			err.Error()))
+	}
+
+	if h.Spec.StorageType == "" {
+		allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+			h.Name,
+			"StorageType can not be empty"))
+	} else {
+		if h.Spec.StorageType != olddbapi.StorageTypeDurable && h.Spec.StorageType != olddbapi.StorageTypeEphemeral {
+			allErr = append(allErr, field.Invalid(field.NewPath("spec").Child("storageType"),
+				h.Name,
+				"StorageType should be either durable or ephemeral"))
+		}
+	}
+
+	return allErr
+}
+
+func (w *HazelcastCustomWebhook) hazelcastValidateVolumes(podTemplate *ofst.PodTemplateSpec) error {
+	if podTemplate == nil {
+		return nil
+	}
+	if podTemplate.Spec.Volumes == nil {
+		return nil
+	}
+
+	for _, rv := range hazelcastReservedVolumes {
+		for _, ugv := range podTemplate.Spec.Volumes {
+			if ugv.Name == rv {
+				return errors.New("Can't use a reserve volume name: " + rv)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (w *HazelcastCustomWebhook) hazelcastValidateVersion(h *olddbapi.Hazelcast) error {
+	hzVersion := catalog.HazelcastVersion{}
+	err := w.DefaultClient.Get(context.TODO(), types.NamespacedName{Name: h.Spec.Version}, &hzVersion)
+	if err != nil {
+		return errors.New("version not supported")
+	}
+	return nil
+}
+
+func (w *HazelcastCustomWebhook) hazelcastValidateVolumesMountPaths(podTemplate *ofst.PodTemplateSpec) error {
+	if podTemplate == nil {
+		return nil
+	}
+	if podTemplate.Spec.Containers == nil {
+		return nil
+	}
+
+	for _, rvmp := range hazelcastReservedVolumeMountPaths {
+		containerList := podTemplate.Spec.Containers
+		for i := range containerList {
+			mountPathList := containerList[i].VolumeMounts
+			for j := range mountPathList {
+				if mountPathList[j].MountPath == rvmp {
+					return errors.New("Can't use a reserve volume mount path name: " + rvmp)
+				}
+			}
+		}
+	}
+
+	if podTemplate.Spec.InitContainers == nil {
+		return nil
+	}
+
+	for _, rvmp := range hazelcastReservedVolumeMountPaths {
+		containerList := podTemplate.Spec.InitContainers
+		for i := range containerList {
+			mountPathList := containerList[i].VolumeMounts
+			for j := range mountPathList {
+				if mountPathList[j].MountPath == rvmp {
+					return errors.New("Can't use a reserve volume mount path name: " + rvmp)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (w *HazelcastCustomWebhook) ValidateUpdate(ctx context.Context, old, newObj runtime.Object) (admission.Warnings, error) {
+	db, ok := newObj.(*olddbapi.Hazelcast)
+	if !ok {
+		return nil, fmt.Errorf("expected an Hazelcast object but got %T", newObj)
+	}
+	hazelcastlog.Info("validate update", "name", db.Name)
+
+	allErr := w.ValidateCreateOrUpdate(db)
+	if len(allErr) == 0 {
+		return nil, nil
+	}
+
+	return nil, apierrors.NewInvalid(schema.GroupKind{Group: "kubedb.com", Kind: "Hazelcast"}, db.Name, allErr)
```

Comment:
Always use const in such places.  Change all occurances of "kubedb.com" & "Hazzlecast"

Reply:
Also resolve the conflicts